azure-iot-device/samples/advanced-hub-scenarios/provision_symmetric_key_and_send_telemetry.py

```diff
@@ -0,0 +1,69 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import asyncio
+from azure.iot.device.aio import ProvisioningDeviceClient
+import os
+from azure.iot.device.aio import IoTHubDeviceClient
+from azure.iot.device import Message
+import uuid
+
+
+messages_to_send = 10
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+async def main():
+    async def register_device():
+        provisioning_device_client = ProvisioningDeviceClient.create_from_symmetric_key(
+            provisioning_host=provisioning_host,
+            registration_id=registration_id,
+            id_scope=id_scope,
+            symmetric_key=symmetric_key,
+        )
+
+        return await provisioning_device_client.register()
+
+    results = await asyncio.gather(register_device())
+    registration_result = results[0]
+    print("The complete registration result is")
+    print(registration_result.registration_state)
+
+    print("Will send telemetry from the provisioned device")
+    device_client = IoTHubDeviceClient.create_from_registration_result_and_symmetric_key(
```

Comment:
>create_from_registration_result_and_symmetric_key [](start = 39, length = 49)

This sample doesn't check for a failed registration.  Or does provisioning_device_client.register() raise an exception if provisioning fails?

Reply:
good idea to check the status. In the current state machine it should give an error but I think it is better to check

---
In reply to: [354615024](https://github.com/Azure/azure-iot-sdk-python/pull/407#discussion_r354615024) [](ancestors = 354615024)